app/modules/enduser_setup.c

```diff
@@ -675,6 +721,27 @@ static int enduser_setup_http_handle_credentials(char *data, unsigned short data
   }
 
 
+
+  char *config_str = (char *) ((uint32_t)strstr(&(data[6]), "config="));
+  if(config_str != NULL)
+  {
+    int config_field_len = LITLEN("config=");
+    char *config_str_start = config_str + config_field_len;
+    int config_str_len = enduser_setup_srch_str(config_str_start, "& ");
+    /* Create a new variable for storing the decoded string.
+     * Add one more size(char) to allow for \0 */
+    uint8 decoded_config_str[config_str_len+1];
```

Comment:
Just a note of caution that the stack space is rather limited. It would be safer to `os_malloc()` this buffer (and later `os_free()` it. As-is, someone could smash the stack by submitting a really long config parameter.